core/src/Cuda/Kokkos_Cuda_Instance.hpp

```diff
@@ -17,30 +17,30 @@ namespace Kokkos {
 namespace Impl {
 
 struct CudaTraits {
-  enum : CudaSpace::size_type { WarpSize = 32 /* 0x0020 */ };
-  enum : CudaSpace::size_type {
-    WarpIndexMask = 0x001f /* Mask for warpindex */
-  };
-  enum : CudaSpace::size_type {
-    WarpIndexShift = 5 /* WarpSize == 1 << WarpShift */
-  };
-
-  enum : CudaSpace::size_type {
-    ConstantMemoryUsage = 0x008000 /* 32k bytes */
-  };
-  enum : CudaSpace::size_type {
-    ConstantMemoryCache = 0x002000 /*  8k bytes */
-  };
-  enum : CudaSpace::size_type {
-    KernelArgumentLimit = 0x001000 /*  4k bytes */
-  };
-  enum : CudaSpace::size_type {
-    MaxHierarchicalParallelism = 1024 /* team_size * vector_length */
-  };
+  static constexpr CudaSpace::size_type WarpSize = 32 /* 0x0020 */;
```

Comment:
The problem with this change is that under C++14, these are not inline variables. Any code which takes its address, even indirectly when passing it by reference, will break under C++14 (and be magically fixed under C++17) unless there is a corresponding out of line definition for it.

Reply:
Ah, just read the first comment.  Nevermind.